postgres-testing/src/main/java/org/folio/postgres/testing/PostgresTesterContainer.java

```diff
@@ -34,44 +44,143 @@ public PostgresTesterContainer() {
    * Start the container.
    */
   @Override
-  public void start(String database, String username, String password) {
-    if (postgreSQLContainer != null) {
+  public void start(String database, String username, String password) throws PostgresTesterStartException {
+    if (primary != null) {
       throw new IllegalStateException("already started");
     }
-    postgreSQLContainer = new PostgreSQLContainer<>(dockerImageName)
-        .withDatabaseName(database)
-        .withUsername(username)
-        .withPassword(password)
-        .withStartupTimeout(Duration.ofSeconds(60));
-    postgreSQLContainer.start();
+
+    GenericContainer<?> tempStandby;
+    var replicationUser = "replicator";
+    var primaryHost = "primaryhost";
+    var dataDirectory = "/var/lib/postgresql/standby/";
+    var tempDirectory = "/tmp/standby/";
+    var hostVolume = "/tmp/rmb-standby-" + UUID.randomUUID();
+    var network = Network.newNetwork();
+
+    // In postgres replication the default is for commit to be async. So no configuration is needed, but our
+    // default for tests is to configure it for synchronous commit.
+    boolean configureSynchronousCommit = System.getProperty(POSTGRES_ASYNC_COMMIT) == null;
+
+    try {
+      primary = new PostgreSQLContainer<>(dockerImageName)
+          .withDatabaseName(database)
+          .withUsername(username)
+          .withPassword(password)
+          .withNetwork(network)
+          .withNetworkAliases(primaryHost)
+          .waitingFor(Wait.forLogMessage(".*database system is ready to accept connections.*\\n", READY_MESSAGE_TIMES));
+
+      if (configureSynchronousCommit) {
+        primary.withEnv("PGOPTIONS", "-c synchronous_commit=remote_apply");
```

Comment:
In line 125. The standby name is the default which is `walreciever` but to enable sync you still have to set it and reload.If I set this in PG_OPTIONS the container doesn't start probably because it expects the standby to be available which it isn't yet.